src/selectors/url-state.js

```diff
@@ -256,10 +268,20 @@ export const getProfileName: Selector<string> = createSelector(
         return matchResult[0];
       }
     }
-    return '';
+    return null;
   }
 );
 
+/**
+ * For now only provide a name for a profile if it came from a zip file.
+ */
+export const getProfileName: Selector<string> = createSelector(
+  getProfileNameOrNull,
+  getProfileMeta,
+  (profileNameOrNull, meta) =>
+    profileNameOrNull === null ? formatMetaInfoString(meta) : profileNameOrNull
```

Comment:
I'm not sure about this change. I would call `formatMetaInfoString` in the component rather than the selector, because what we want to do in case of an empty name can be different for each components.

As an example: in the publish action, we store the profile name in the indexeddb store. Then when listing the stored profiles, when the name is absent, we display a sliced profile token ("hash"). With this change we always have a profile name now and so that wouldn't work. This can be alleviated by taking `getProfilenameOrNull` there too, but my point is that deciding what to do when it's absent is the role of the component.

This may look like nitpicking but I think this is more important than that.